dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/AbstractSnowflakeConnector.java

```diff
@@ -88,24 +88,28 @@ public Handle open(@Nonnull ConnectorArguments arguments) throws Exception {
         newDriver(arguments.getDriverPaths(), "net.snowflake.client.jdbc.SnowflakeDriver");
     DataSource dataSource =
         new SimpleDriverDataSource(driver, url, arguments.getUser(), arguments.getPassword());
-    return checkCurrentDatabaseExists(arguments, new JdbcHandle(dataSource));
+    JdbcHandle jdbcHandle = new JdbcHandle(dataSource);
+    if (!arguments.getDatabases().isEmpty()) {
+      checkCurrentDatabaseExists(arguments.getDatabases().get(0), jdbcHandle.getJdbcTemplate());
```

Comment:
The method name `checkCurrentDatabaseExists` starts with `check`, even though it issues `USE DATABASE` that alters the JDBC session (i.e. it's not just a check). I would suggest to either rename the method or call 2 methods - one for checking that the database exists, another for setting the current database for the session.